include/triton/core/tritonserver.h

```diff
@@ -2267,6 +2267,10 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_MetricFamilyNew(
     const char* name, const char* description);
 
 /// Delete a metric family object.
+/// A TRITONSERVER_MetricFamily* object should be deleted AFTER its
+/// corresponding TRITONSERVER_Metric* objects have been deleted.
+/// If a family is deleted before its metrics, an error will be
```

Comment:
If the family has been deleted, what happened if user calls other `TRITONSERVER_MetricXxx` functions? Should also return error as the metric will no longer be valid?

Reply:
Adding more unit tests around this

Reply:
Technically MetricSet/Increment/Value etc. will work as they operate directly on the metric and don't need to communicate with tc::MetricFamily object. 

Since we don't remove a MetricFamily's prometheus::Family from the prometheus::Registry on ~MetricFamily destruction, the tc::Metric will have a valid reference to the prometheus::Metric still in the prometheus::registry and can update it accordingly.

If we want to disallow this, we can also invalidate the reference by setting `tc::Metric::metric_ = nullptr`? Let me know what you think.

Reply:
I also added a similar check to MetricFamilyDelete to instead enforce that FamilyDelete is called only once its dependent metrics are deleted.